snuba/datasets/discover.py

```diff
@@ -145,7 +160,11 @@ def detect_table(
                 transaction_columns.add(col.column_name)
 
         event_mismatch = event_columns and selected_table == TRANSACTIONS
```

Comment:
shouldn't it be `selected_table in [TRANSACTIONS, EVENTS_AND_TRANSACTIONS]` ?

Reply:
I don't think so... since we have today:
EVENTS -> events table
EVENTS_AND_TRANSACTIONS -> events table still (merge table in future)
TRANSACTIONS -> transactions table
So I think the only event mismatch today arises when the selected table is transactions but there are event columns in the request